lib/AST/Decl.cpp

```diff
@@ -4707,6 +4696,23 @@ Type SubscriptDecl::getElementInterfaceType() const {
   return elementTy->castTo<AnyFunctionType>()->getResult();
 }
 
+void SubscriptDecl::computeType() {
+  auto &ctx = getASTContext();
+
+  auto elementTy = getElementTypeLoc().getType();
+  auto indicesTy = getIndices()->getInterfaceType(ctx);
```

Comment:
In future, what's the right way to convert these kinds of things to speak in terms of params?

Reply:
Add a way to go from a ParamDecl to an AnyFunctionType::Param. Then we can remove ParameterList::getInterfaceType().